qiskit/providers/ibmq/api/clients/websocket.py

```diff
@@ -161,75 +180,119 @@ def get_job_status(
+                # Read messages from the server until the connection is closed or
+                # a timeout has been reached.
+                while True:
+                    try:
+                        with warnings.catch_warnings():
+                            # Suppress websockets deprecation warnings until the fix is available
+                            warnings.filterwarnings("ignore", category=DeprecationWarning)
+                            if timeout:
+                                response_raw = yield from asyncio.wait_for(
+                                    websocket.recv(), timeout=timeout)
+
+                                # Decrease the timeout, with a 5-second grace period.
+                                elapsed_time = time.time() - start_time
+                                timeout = max(5, int(original_timeout - elapsed_time))
+                            else:
+                                response_raw = yield from websocket.recv()
+                        logger.debug('Received message from websocket: %s',
+                                     response_raw)
+
+                        # Successfully received a message, reset retry counter.
+                        current_retry_attempt = 0
+
+                        response = WebsocketMessage.from_bytes(response_raw)
+                        last_status = response.data
+
+                        job_status = response.data.get('status')
+                        if (job_status and
+                                ApiJobStatus(job_status) in API_JOB_FINAL_STATES):
+                            break
+
+                    except futures.TimeoutError:
+                        # Timeout during our wait.
+                        raise WebsocketTimeoutError('Timeout reached') from None
+                    except ConnectionClosed as ex:
+                        # From the API:
+                        # 4001: closed due to an internal errors
+                        # 4002: closed on purpose (no more updates to send)
+                        # 4003: closed due to job not found.
+                        message = 'Unexpected error'
+                        if ex.code == 4001:
+                            message = 'Internal server error'
+                        elif ex.code == 4002:
+                            break
+                        elif ex.code == 4003:
+                            attempt_retry = False  # No point in retrying.
+                            message = 'Job id not found'
+
+                        raise WebsocketError('Connection with websocket closed '
+                                             'unexpectedly: {}(status_code={})'
+                                             .format(message, ex.code)) from ex
+
+            # Errors unrelated to websocket connection, not worth retrying.
+            except (WebsocketIBMQProtocolError, WebsocketTimeoutError) as ex:
+                raise ex
```

Comment:
You don't need to `except` the errors if you're just going to raise it? Or are you trying to document the possible exceptions?

Reply:
I agree! I think it would be much cleaner to simply remove them, since they are going to be raised regardless.